recipes/igblast/build.sh

```diff
@@ -1,45 +1,71 @@
 #!/bin/bash
-set -euo pipefail
+set -xeuo pipefail
+
+# This script uses ideas from the build script for BLAST. See comments there.
 
-IGBLAST_ADDRESS=ftp://ftp.ncbi.nih.gov/blast/executables/igblast/release
 SHARE_DIR=$PREFIX/share/igblast
 
 mkdir -p $PREFIX/bin
 
-if [ $(uname) == Linux ]; then
-  # The binaries want libbz2.so.1, but the correct soname is libbz2.so.1.0
-  for name in makeblastdb igblastn igblastp; do
-    patchelf --replace-needed libbz2.so.1 libbz2.so.1.0 bin/$name
-  done
-fi
-
 # $SHARE_DIR contains the actual igblastn and igblastp binaries and also the
 # required data files. Wrappers will be installed into $PREFIX/bin that set
 # $IGDATA to point to those data files.
 mkdir -p $SHARE_DIR/bin
 
-# Copy binaries and wrappers
-for name in igblastn igblastp; do
-  mv bin/$name $SHARE_DIR/bin/
-  sed "s/igblastn/$name/g" $RECIPE_DIR/igblastn.sh > $PREFIX/bin/$name
-  chmod +x $PREFIX/bin/$name
-done
+if [[ $(uname) == Linux ]]; then
+    export CFLAGS="$CFLAGS -O2"
+    export CXXFLAGS="$CXXFLAGS -O2"
+    export CPPFLAGS="$CPPFLAGS -I$PREFIX/include -I$PREFIX/include/ncbi-vdb"
+    export LDFLAGS="$LDFLAGS -L$PREFIX/lib"
+    export CC_FOR_BUILD=$CC
+    export AR="$AR rcs"
 
-# No wrapper needed
-mv bin/makeblastdb $PREFIX/bin/
-
-wget $IGBLAST_ADDRESS/edit_imgt_file.pl
-# Replace the hardcoded perl shebang pointing to /opt with `#!/usr/bin/env perl`.
-sed -i.backup '1 s_^.*$_#!/usr/bin/env perl_' edit_imgt_file.pl
-chmod +x edit_imgt_file.pl
-mv edit_imgt_file.pl $PREFIX/bin/
+    cd c++
 
+    # IgBLAST is based on the BLAST source code and building it produces
+    # a similar set of shared libraries if --with-dll is used. To avoid
+    # conflicts when installing IgBLAST and BLAST simultaneously,
+    # we link IgBLAST statically.
+    ./configure.orig \
+        --with-static-exe \
```

Comment:
Was `/lib64/libc.so.6: version `GLIBC_2.14' not found` the only error you encountered without this?
I'm a little surprised that you got that at all when building from source because it indicates that some other binaries built with a foreign glibc are present.

Reply:
Hm, the CI test logs don’t seem to be available anymore, so I cannot prove it, but I’m certain this wasn’t the only problem I encountered. I also did a lot of local testing, and perhaps this was a message came from one of those test runs.

Also, the build system by default downloads a couple of dependencies that it needs, which I had to disable. It could also be that the error came from one of those.

Reply:
Ah, the old logs aren’t linked here because I force-pushed the branch. They are [still available](https://circleci.com/gh/bioconda/bioconda-recipes/tree/igblast). If you want, I can dig further.

Reply:
Ah, I remember now: I got `lib64/libc.so.6: version GLIBC_2.14` when trying to package the binary version. That was the reason I had to switch to compiling from sources.

Reply:
> They are [still available](https://circleci.com/gh/bioconda/bioconda-recipes/tree/igblast).

Perfect. I took a look and we got

<details><summary>

```
$ curl -sL 'https://circleci.com/api/v1.1/project/github/bioconda/bioconda-recipes/62852/output/106/0?file=true' | grep -oE -- '-o lib[^ ]*\.so' | sort
```
</summary>

```
-o libaccess.so
-o libalign_format.so
-o libbiblio.so
-o libbiotree.so
-o libblastdb_format.so
-o libblastdb.so
-o libblastinput.so
-o libblast_services.so
-o libblast.so
-o libblast_sra_input.so
-o libblastxml2.so
-o libblastxml.so
-o libcdd.so
-o libcn3d.so
-o libcomposition_adjustment.so
-o libconnect.so
-o libconnssl.so
-o libcreaders.so
-o libdbapi_driver.so
-o libdbapi.so
-o libdocsum.so
-o libentrez2cli.so
-o libentrez2.so
-o libentrezgene.so
-o libfeatdef.so
-o libgbseq.so
-o libgene_info.so
-o libgeneral.so
-o libgenesbyloc.so
-o libgenome_collection.so
-o libhomologene.so
-o libid1cli.so
-o libid1.so
-o libid2cli.so
-o libid2.so
-o libid2_split.so
-o libigblast.so
-o libinsdseq.so
-o liblmdb.so
-o libmacro.so
-o libmedlars.so
-o libmedline.so
-o libmim.so
-o libmlacli.so
-o libmla.so
-o libmmdb.so
-o libncbimime.so
-o libncbi_xblobstorage_netcache.so
-o libncbi_xcache_dbapi.so
-o libncbi_xcache_netcache.so
-o libncbi_xloader_blastdb_rmt.so
-o libncbi_xloader_blastdb.so
-o libncbi_xloader_genbank.so
-o libncbi_xreader_cache.so
-o libncbi_xreader_id1.so
-o libncbi_xreader_id2.so
-o libncbi_xreader.so
-o libobjcoords.so
-o libobjprt.so
-o libomssa.so
-o libpcassay.so
-o libpcsubstance.so
-o libproj.so
-o libproteinkmer.so
-o libpubmed.so
-o libpub.so
-o libregexp.so
-o libremapcli.so
-o libremap.so
-o libscoremat.so
-o libseqcode.so
-o libseqdb.so
-o libseqedit.so
-o libseqmasks_io.so
-o libseqset.so
-o libseq.so
-o libseqsplit.so
-o libseqtest.so
-o libsequtil.so
-o libsraread.so
-o libsubmit.so
-o libtables.so
-o libtaxon1.so
-o libtaxon3.so
-o libtest_mt.so
-o libtinyseq.so
-o libvalerr.so
-o libvalid.so
-o libvariation.so
-o libvarrep.so
-o libwritedb.so
-o libxalgoalignutil.so
-o libxalgoblastdbindex.so
-o libxalgodustmask.so
-o libxalgoseq.so
-o libxalgowinmask.so
-o libxalnmgr.so
-o libxblastformat.so
-o libxblast.so
-o libxcgi.so
-o libxcleanup.so
-o libxcompress.so
-o libxconnect.so
-o libxconnserv.so
-o libxcser.so
-o libxfcgi.so
-o libxformat.so
-o libxhtml.so
-o libxlogging.so
-o libxncbi.so
-o libxnetblastcli.so
-o libxnetblast.so
-o libxobjedit.so
-o libxobjmgr.so
-o libxobjreadex.so
-o libxobjread.so
-o libxobjsimple.so
-o libxobjutil.so
-o libxqueryparse.so
-o libxregexp.so
-o libxregexp_template_tester.so
-o libxser.so
-o libxthrserv.so
-o libxutil.so
-o libxxconnect.so
```
</details>
in there, meaning there are quite a lot shared libraries that get built and would have to be moved to `${PREFIX}/lib` (maybe/likely creating path conflicts with other packages).
We could move those under `share/igblast/lib` and use `export LDFLAGS="-Wl,-rpath,${SHARE_DIR}/lib ${LDFLAGS}"`.
But it's probably not really worth to do as the official binaries also statically link to those libraries. So, all good as is.

> Ah, I remember now: I got `lib64/libc.so.6: version GLIBC_2.14` when trying to package the binary version.

Alright, then it all makes sense again :).